File: Backend/MyApp.CoreService/Models/Account.cs
using MyApp.CoreService.Enums;

namespace MyApp.CoreService.Models;

public class Account
{
    public int Id { get; set; }
    public int OwnerId { get; set; }
    public decimal Balance { get; set; } = 0m;
    public string Currency { get; set; } = "RUB";
    public AccountStatus Status { get; set; } = AccountStatus.Open;
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset? ClosedAt { get; set; }

    public ICollection<Transaction> Transactions { get; set; } = new List<Transaction>();
}

File: Mobile/components/account/src/main/java/nekit/corporation/loan_shared/domain/model/Account.kt
package nekit.corporation.loan_shared.domain.model

import nekit.corporation.loan_shared.data.datasource.remote.model.AccountStatus
import java.time.OffsetDateTime

data class Account(
    val id: Int,
    val ownerId: Int,
    val balance: Double,
    val currency: String,
    val status: AccountStatus,
    val createdAt: OffsetDateTime,
    val closedAt: OffsetDateTime? = null
)
